hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/JpaMetamodelImpl.java

```diff
@@ -493,6 +500,34 @@ public void processJpa(
 			for ( EmbeddableDomainType<?> embeddable : context.getEmbeddableTypeSet() ) {
 				this.jpaEmbeddableDescriptorMap.put( embeddable.getJavaType(), embeddable );
 			}
+			Stream.concat(
+					context.getEntityTypesByEntityName().values().stream(),
+					Stream.concat(
+							context.getMappedSuperclassTypeMap().values().stream(),
+							context.getEmbeddableTypeSet().stream()
+					)
+			).forEach( managedDomainType -> {
```

Comment:
We might consider the following pattern:
```
Stream.of( 
					context.getEntityTypesByEntityName().values().stream(),
					context.getMappedSuperclassTypeMap().values().stream(),
					context.getEmbeddableTypeSet().stream()
			).flatMap( Function.identity() ).forEach( ... )
```
As per https://www.techempower.com/blog/2016/10/19/efficient-multiple-stream-concatenation-in-java/, the flatmap approach performs better when the stream sizes are not large.

Reply:
According to the article the suggestion is to use `flatMap(Stream::concat)` so I'd rather stick to this pattern.